File: backend/src/main/java/com/demo/travel_expense_management/reimbursement/service/ReimbursementService.java
package com.demo.travel_expense_management.reimbursement.service;

import com.demo.travel_expense_management.reimbursement.dto.request.RegisterReimbursementRequest;
import com.demo.travel_expense_management.reimbursement.dto.response.ReimbursementResponse;

public interface ReimbursementService {

    ReimbursementResponse registerReimbursement(Long legalizationId, RegisterReimbursementRequest request);
}

File: frontend/src/lib/services/reimbursement.service.ts
import { apiClient } from '$lib/api/api-client';
import type {
	ReimbursementResponseDto,
	RegisterReimbursementDto,
} from '$lib/dto/reimbursement';

class ReimbursementService {
	private readonly resourcePath = '/v1/reimbursements';

	registerReimbursement(
		legalizationId: number,
		payload: RegisterReimbursementDto,
	): Promise<ReimbursementResponseDto> {
		return apiClient.post<ReimbursementResponseDto>(
			`${this.resourcePath}/legalization/${legalizationId}`,
			payload,
		);
	}
}

export const reimbursementService = new ReimbursementService();
